src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -323,6 +382,13 @@ private string GetTemplateParameterErrorsMessage(ITemplateInfo templateInfo, out
             return null;
         }
 
+        private string GetLanguageMismatchErrorMessage(string inputLanguage)
+        {
+            string invalidLanguageErrorText = LocalizableStrings.InvalidTemplateParameterValues;
+            invalidLanguageErrorText += Environment.NewLine + string.Format(LocalizableStrings.InvalidParameterDetail, "-lang", inputLanguage, "--language");
```

Comment:
I did it this way to be more consistent with the output from GetTemplateParameterErrorsMessage(), which lists the canonical (full name) at the end of the message. Perhaps a quick discussion offline is best here.